xrayvision/tests/test_clean.py

```diff
@@ -16,18 +16,26 @@ def test_hogbom_intensity_change(self):
         pos1 = [15, 30]
         pos2 = [40, 32]
         # Creating a "clean" map as a base with 2 point sources
-        clean_map = np.zeros((N, M), dtype=complex)
+        clean_map = np.zeros((N, M))
         clean_map[pos1[0], pos1[1]] = 1.
         clean_map[pos2[0], pos2[1]] = 0.8
 
         u, v = np.meshgrid(np.arange(N), np.arange(M))
         uv_in = np.array([u, v]).reshape(2, N * M)
         vis_in = np.zeros(N * M, dtype=complex)
 
+        dirty_beam = np.zeros((N, M))
+        dirty_beam[(N-1)//4:(N-1)//4 + (N-1)//2, (M-1)//2] = 0.75
+        dirty_beam[(N-1)//2, (M-1)//4:(M-1)//4 + (M-1)//2, ] = 0.75
+        dirty_beam[(N-1)//2, (M-1)//2] = 1.0
+
+        dirty_map = signal.convolve2d(clean_map, dirty_beam, mode='same')
+
         vis = Visibility(uv_in, vis_in)
```

Comment:
When you use all then u, v components it is not a dirty map it's basically Fourier transform of the dirty map. If you want to do it from visibilities you need to define some subset of all the visibilities and from those do a back project to obtain the dirty map to use in line 32. So something like this

```
vis_in = np.zero((N, M), dtype=complex)
vis_in[(N-1)//4:(N-1)//4 + (N-1)//2, (M-1)//2] = 0.75
vis_in[(N-1)//2, (M-1)//4:(M-1)//4 + (M-1)//2, ] = 0.75
vis_in[(N-1)//2, (M-1)//2] = 1.0
vis_in = vis_in.reshap(N*M)
```

I'm not sure what shape uv_in is but you'd have to pick out the corresponding elements in that array as well.